Source/OxyPlot/Series/BarSeries/RectangleBarSeries.cs

```diff
@@ -152,15 +157,25 @@ public override void Render(IRenderContext rc)
                 return;
             }
 
-            var clippingRect = this.GetClippingRect();
+			// determine render range
+			var xmin = this.XAxis.ActualMinimum;
+			var xmax = this.XAxis.ActualMaximum;
+			this.winIndex = FindBarIndex(Items, xmin, this.winIndex);
+
+	        if (this.winIndex > 0)
+	        {
+		        this.winIndex--;
+	        }
 
-            int i = 0;
+			var clippingRect = this.GetClippingRect();
```

Comment:
Sorry for that, I didn't intend for RectangleBarSeries code to get into this pull request at all — I guess, GitHub adds latest commits to an open pull request automatically. That was only the first commit and RectangleBarSeries needs more work.